### ISSUE TITLE
Large sets of collections in the `Move` modal takes a long time to load or doesn't work

### ISSUE THREAD

**Issue by oisincoveney** [metabase/metabase#30117]
**Describe the bug**
Trying to select a collection (when trying to move a dashboard/question from the `Move` dropdown) from a huge set of collections either takes a long time to load, or causes the application to run out of memory and crash.

**To Reproduce**
1) Loaded ~75k collections into my local environment using 
```
for i in {1..100000}
do
curl --location '[http://localhost:3000/api/collection](http://localhost:3000/api/collection)' \
--header 'X-Metabase-Session: '<USER KEY>' \
--header 'Content-Type: application/json' \
--data '{
"parent_id": 13,
"authority_level": null,
"color": "#509EE3",
"description": null,
"name": "big boi '$i'"
}'
done
```

2) Try to create a question, and move the question to an existing collection using the dialog.

**Expected behavior**
Collections should load relatively quickly (ideally, instantaneously) so the user can navigate through all of the collections.

**Screenshots/videos**

https://user-images.githubusercontent.com/25306947/232076960-7569e3f9-2161-4ea0-adaa-5fe28a425203.mov

https://user-images.githubusercontent.com/25306947/232077029-7357629c-8b5d-408c-a8f7-f2fbf263adc7.mov

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "20+36",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "20",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "20+36",
    "os.name": "Mac OS X",
    "os.version": "13.2.1",
    "user.language": "en",
    "user.timezone": "Europe/Tallinn"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "bigquery-cloud-sdk"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "dev",
    "version": {
      "date": "2023-04-12",
      "src_hash": "79fe0a9a6ab5333c2a36a273e169222fe5c3f19b",
      "tag": "v0.45.1-SNAPSHOT",
      "branch": "29971-fe-editing-a-dashboard",
      "hash": "18e97ea"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```
**Severity**
Depending on the size of a collection set a company has, this issue can become severe and cause the application to stop working.

**Comment by ranquild** [comment]
Would require re-implementing `ItemPicker` using `search` endpoint. We would load items for a selected collection only instead of loading the entire collection tree. Would also require some BE changes, like adding `collection` to `here` and `below` properties so we could know whether a collection has subcollections.

**Comment by iethree** [comment]
this will likely be addressed by #35564
